### ISSUE TITLE
[bitnami/clickhouse] metrics for prometheus doesn't work

### ISSUE THREAD

**Issue by zhenzh** [bitnami/charts#14109]
### Name and Version

bitnami/clickouse 2.2.1

### What steps will reproduce the bug?

The port 8001 specified in helm for prometheus exporter doesn't expose any metrics data after enabled metrics

### What do you see instead?

Metrics for clickhouse can be successfully scrapped by prometheus.

**Comment by fmulero** [comment]
Hi Harry

Could you provide the steps to reproduce it? I've just run the following and I can see the metrics:
```console
$ helm install clickhouse bitnami/clickhouse --set metrics.enabled=true
$ kubectl run curl-metrics -it --image bitnami/bitnami-shell --restart=Never -- curl http://clickhouse:8001/metrics

# HELP ClickHouseProfileEvents_Query Number of queries to be interpreted and potentially executed. Does not include queries that failed to parse or were rejected due to AST size limits, quota limits or limits on the number of simultaneously running queries. May include internal queries initiated by ClickHouse itself. Does not count subqueries.
# TYPE ClickHouseProfileEvents_Query counter
ClickHouseProfileEvents_Query 0
# HELP ClickHouseProfileEvents_SelectQuery Same as Query, but only for SELECT queries.
# TYPE ClickHouseProfileEvents_SelectQuery counter
ClickHouseProfileEvents_SelectQuery 0
# HELP ClickHouseProfileEvents_InsertQuery Same as Query, but only for INSERT queries.
# TYPE ClickHouseProfileEvents_InsertQuery counter
ClickHouseProfileEvents_InsertQuery 0
# HELP ClickHouseProfileEvents_AsyncInsertQuery Same as InsertQuery, but only for asynchronous INSERT queries.
# TYPE ClickHouseProfileEvents_AsyncInsertQuery counter
ClickHouseProfileEvents_AsyncInsertQuery 0
# HELP ClickHouseProfileEvents_AsyncInsertBytes Data size in bytes of asynchronous INSERT queries.
# TYPE ClickHouseProfileEvents_AsyncInsertBytes counter
ClickHouseProfileEvents_AsyncInsertBytes 0
# HELP ClickHouseProfileEvents_FailedQuery Number of failed queries.
# TYPE ClickHouseProfileEvents_FailedQuery counter
ClickHouseProfileEvents_FailedQuery 0
# HELP ClickHouseProfileEvents_FailedSelectQuery Same as FailedQuery, but only for SELECT queries.
# TYPE ClickHouseProfileEvents_FailedSelectQuery counter
ClickHouseProfileEvents_FailedSelectQuery 0
# HELP ClickHouseProfileEvents_FailedInsertQuery Same as FailedQuery, but only for INSERT queries.
# TYPE ClickHouseProfileEvents_FailedInsertQuery counter
ClickHouseProfileEvents_FailedInsertQuery 0
# HELP ClickHouseProfileEvents_FailedAsyncInsertQuery Number of failed ASYNC INSERT queries.
# TYPE ClickHouseProfileEvents_FailedAsyncInsertQuery counter
ClickHouseProfileEvents_FailedAsyncInsertQuery 0
....
```

**Comment by zhenzh** [comment]
Sorry for late response. I have tried in my env with relative simple config again but still cannot read metrics for port 8001.

Here's my steps:
```
helm install clickhouse-test bitnami/clickhouse -f - <<EOF
persistence:
  enabled: false
metrics:
  enabled: true
replicaCount: 1
zookeeper:
  enabled: false
externalZookeeper:
  servers: ["zookeeper"]
EOF

NAME: clickhouse-test
LAST DEPLOYED: Thu Jan 12 13:41:40 2023
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
CHART NAME: clickhouse
CHART VERSION: 2.2.4
APP VERSION: 22.12.3

** Please be patient while the chart is being deployed **

ClickHouse is available in the following address:

    kubectl port-forward --namespace default svc/clickhouse-test 9000:9000 &

Credentials:

    echo "Username      : default"
    echo "Password      : $(kubectl get secret --namespace default clickhouse-test -o jsonpath="{.data.admin-password}" | base64 -d)"

$ kubectl get pod
NAME                                     READY   STATUS    RESTARTS   AGE
clickhouse-test-shard0-0                 1/1     Running   0          31m
clickhouse-test-shard1-0                 1/1     Running   0          31m

$ kubectl get svc
NAME                                             TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                                                 AGE
clickhouse-test                                  ClusterIP   10.247.213.24    <none>        8123/TCP,9000/TCP,9004/TCP,9005/TCP,9009/TCP,8001/TCP   32m
clickhouse-test-headless                         ClusterIP   None             <none>        8123/TCP,9000/TCP,9004/TCP,9005/TCP,9009/TCP            32m

$ kubectl exec -it elasticsearch-kibana-674b584964-fsqz9 -- sh
$ curl clickhouse-test:8001/metrics
curl: (7) Failed to connect to clickhouse-test port 8001: Connection refused
$ curl clickhouse-test:8123
Ok.
```

**Comment by javsalgar** [comment]
Hi,

Could you try executing `helm repo update` and then execute the same steps as @fmulero?

**Comment by zhenzh** [comment]
Here's is.

Since pv cannot be auto-created in my env which pends installation, I set persistence disabled.

```
$ helm install clickhouse bitnami/clickhouse --set metrics.enabled=true --set persistence.enabled=false --set zookeeper.persistence.enabled=false
NAME: clickhouse
LAST DEPLOYED: Thu Jan 12 17:21:30 2023
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
CHART NAME: clickhouse
CHART VERSION: 2.2.4
APP VERSION: 22.12.3

** Please be patient while the chart is being deployed **

ClickHouse is available in the following address:

    kubectl port-forward --namespace default svc/clickhouse 9000:9000 &

Credentials:

    echo "Username      : default"
    echo "Password      : $(kubectl get secret --namespace default clickhouse -o jsonpath="{.data.admin-password}" | base64 -d)"

$ kubectl get pod
NAME                     READY   STATUS    RESTARTS   AGE
clickhouse-shard0-0      2/2     Running   0          16m
clickhouse-shard0-1      2/2     Running   0          16m
clickhouse-shard0-2      2/2     Running   0          16m
clickhouse-shard1-0      2/2     Running   0          16m
clickhouse-shard1-1      2/2     Running   0          16m
clickhouse-shard1-2      2/2     Running   0          16m
clickhouse-zookeeper-0   2/2     Running   0          16m
clickhouse-zookeeper-1   2/2     Running   0          16m
clickhouse-zookeeper-2   2/2     Running   0          16m

$ kubectl get svc
NAME                            TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)                                                 AGE
clickhouse                      ClusterIP   10.247.201.23   <none>        8123/TCP,9000/TCP,9004/TCP,9005/TCP,9009/TCP,8001/TCP   22m
clickhouse-headless             ClusterIP   None            <none>        8123/TCP,9000/TCP,9004/TCP,9005/TCP,9009/TCP            22m
clickhouse-zookeeper            ClusterIP   10.247.96.47    <none>        2181/TCP,2888/TCP,3888/TCP                              22m
clickhouse-zookeeper-headless   ClusterIP   None            <none>        2181/TCP,2888/TCP,3888/TCP                              22m

$ kubectl run curl-metrics -it --image bitnami/bitnami-shell --restart=Never -- curl http://clickhouse:8001/metrics
error: timed out waiting for the condition
HarrydeMacBook-Pro:Downloads harry$ kubectl run curl-metrics -it --image bitnami/bitnami-shell --restart=Never -- curl http://clickhouse:8001/metrics
Error from server (AlreadyExists): pods "curl-metrics" already exists
$ kubectl logs curl-metrics
curl: (7) Failed to connect to clickhouse port 8001: Connection refused
```

**Comment by zhenzh** [comment]
Here's is.

Since pv cannot be auto-created in my env which pends installation, I set persistence disabled.

```
$ helm install clickhouse bitnami/clickhouse --set metrics.enabled=true --set persistence.enabled=false --set zookeeper.persistence.enabled=false
NAME: clickhouse
LAST DEPLOYED: Thu Jan 12 17:21:30 2023
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
CHART NAME: clickhouse
CHART VERSION: 2.2.4
APP VERSION: 22.12.3

** Please be patient while the chart is being deployed **

ClickHouse is available in the following address:

    kubectl port-forward --namespace default svc/clickhouse 9000:9000 &

Credentials:

    echo "Username      : default"
    echo "Password      : $(kubectl get secret --namespace default clickhouse -o jsonpath="{.data.admin-password}" | base64 -d)"

$ kubectl get pod
NAME                     READY   STATUS    RESTARTS   AGE
clickhouse-shard0-0      2/2     Running   0          16m
clickhouse-shard0-1      2/2     Running   0          16m
clickhouse-shard0-2      2/2     Running   0          16m
clickhouse-shard1-0      2/2     Running   0          16m
clickhouse-shard1-1      2/2     Running   0          16m
clickhouse-shard1-2      2/2     Running   0          16m
clickhouse-zookeeper-0   2/2     Running   0          16m
clickhouse-zookeeper-1   2/2     Running   0          16m
clickhouse-zookeeper-2   2/2     Running   0          16m

$ kubectl get svc
NAME                            TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)                                                 AGE
clickhouse                      ClusterIP   10.247.201.23   <none>        8123/TCP,9000/TCP,9004/TCP,9005/TCP,9009/TCP,8001/TCP   22m
clickhouse-headless             ClusterIP   None            <none>        8123/TCP,9000/TCP,9004/TCP,9005/TCP,9009/TCP            22m
clickhouse-zookeeper            ClusterIP   10.247.96.47    <none>        2181/TCP,2888/TCP,3888/TCP                              22m
clickhouse-zookeeper-headless   ClusterIP   None            <none>        2181/TCP,2888/TCP,3888/TCP                              22m

$ kubectl run curl-metrics -it --image bitnami/bitnami-shell --restart=Never -- curl http://clickhouse:8001/metrics
error: timed out waiting for the condition
HarrydeMacBook-Pro:Downloads harry$ kubectl run curl-metrics -it --image bitnami/bitnami-shell --restart=Never -- curl http://clickhouse:8001/metrics
Error from server (AlreadyExists): pods "curl-metrics" already exists
$ kubectl logs curl-metrics
curl: (7) Failed to connect to clickhouse port 8001: Connection refused
```

**Comment by zhenzh** [comment]
Here's is.

Since pv cannot be auto-created in my env which pends installation, I set persistence disabled.

```
$ helm install clickhouse bitnami/clickhouse --set metrics.enabled=true --set persistence.enabled=false --set zookeeper.persistence.enabled=false
NAME: clickhouse
LAST DEPLOYED: Thu Jan 12 17:21:30 2023
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
CHART NAME: clickhouse
CHART VERSION: 2.2.4
APP VERSION: 22.12.3

** Please be patient while the chart is being deployed **

ClickHouse is available in the following address:

    kubectl port-forward --namespace default svc/clickhouse 9000:9000 &

Credentials:

    echo "Username      : default"
    echo "Password      : $(kubectl get secret --namespace default clickhouse -o jsonpath="{.data.admin-password}" | base64 -d)"

$ kubectl get pod
NAME                     READY   STATUS    RESTARTS   AGE
clickhouse-shard0-0      2/2     Running   0          16m
clickhouse-shard0-1      2/2     Running   0          16m
clickhouse-shard0-2      2/2     Running   0          16m
clickhouse-shard1-0      2/2     Running   0          16m
clickhouse-shard1-1      2/2     Running   0          16m
clickhouse-shard1-2      2/2     Running   0          16m
clickhouse-zookeeper-0   2/2     Running   0          16m
clickhouse-zookeeper-1   2/2     Running   0          16m
clickhouse-zookeeper-2   2/2     Running   0          16m

$ kubectl get svc
NAME                            TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)                                                 AGE
clickhouse                      ClusterIP   10.247.201.23   <none>        8123/TCP,9000/TCP,9004/TCP,9005/TCP,9009/TCP,8001/TCP   22m
clickhouse-headless             ClusterIP   None            <none>        8123/TCP,9000/TCP,9004/TCP,9005/TCP,9009/TCP            22m
clickhouse-zookeeper            ClusterIP   10.247.96.47    <none>        2181/TCP,2888/TCP,3888/TCP                              22m
clickhouse-zookeeper-headless   ClusterIP   None            <none>        2181/TCP,2888/TCP,3888/TCP                              22m

$ kubectl run curl-metrics -it --image bitnami/bitnami-shell --restart=Never -- curl http://clickhouse:8001/metrics
error: timed out waiting for the condition
HarrydeMacBook-Pro:Downloads harry$ kubectl run curl-metrics -it --image bitnami/bitnami-shell --restart=Never -- curl http://clickhouse:8001/metrics
Error from server (AlreadyExists): pods "curl-metrics" already exists
$ kubectl logs curl-metrics
curl: (7) Failed to connect to clickhouse port 8001: Connection refused
```

**Comment by marcosbc** [comment]
Hi @zhenzh, have you tried to `kubectl exec` into a ClickHouse pod, and then `curl 127.0.0.1:8001`?

Another thing. It could be possible that port 8001 is actually a typo, as according to the [docs](https://clickhouse.com/docs/en/guides/sre/network-ports/) it should be port 9363. If so, could you try `curl 127.0.0.1:9363` as well?

If it is a typo, feel free to contribute the fix as a PR, we would be glad to help with the review and release of the changes.

**Comment by zhenzh** [comment]
Neither 127.0.0.1:8001 nor curl 127.0.0.1:9363 woks for me.

I have tried deploy the clickhouse chart in my docker-desktop k8s env from my laptop. And I found the metrics works.
Only the clickhouse deployed in managed k8s provided by my cloud provider doesn't work.

**Comment by zhenzh** [comment]
I have also tried changing metrics port to other range like 9xxx in my managed k8s to get ride of the security policy impacts, the issue remains.
I have also installed lots of other bitnami charts in the same k8s, such as elasticsearch, kafka, redis, postgresql and etc. All these applications have no metrics issue.

**Comment by marcosbc** [comment]
Hi @zhenzh, after checking this, I've been able to get it working with `--set metrics.enabled=true`. Then, inside the running container, the metrics are available at `127.0.0.1:8001/metrics`.

Note that port 8001 is also what is set as `ports.metrics` in `values.yaml`, so it should be properly configured.
